### PR TITLE
Implement Support for ERC-4337 EntryPoint v0.7

### PR DESCRIPTION
**Pull request by mmv08**
**This PR**:
- Addresses the M-01 from the audit by updating the ERC-4337 contracts to the latest ones from the development branch (presumably, they will be released as version 0.7.0). The most notable change is the new `PackedUserOperation` struct with the new packed gas limit field `accountGasLimits` that consists of two `uint128` packed values `(validationGasLimit, callGasLimit)`
- Implements https://github.com/safe-global/safe-modules/issues/225

Forked contracts repo with build artifacts: https://github.com/5afe/account-abstraction
More about 0.7.0 changes: https://github.com/safe-global/safe-modules/issues/215

**Issues/concerns**:
- Using unaudited non-release candidate contract versions
- The reference bundler is still in development, thus e2e tests are failing (Confirmed with ERC4337 team)

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -1 +1,2 @@
 modules/**
```
Comment (mmv08) [inline .prettierignore:1]: This and other formatting/linting-related changes were needed to fix the failing CI job. The problem was conflicting eslint-prettier-plugin rules and prettier itself due to misconfiguration.

TLDR: For linting and formatting javascript/typescript code, we use [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier), which runs prettier as ESLint rules. For checking formatting, we use a standalone `prettier` package instead. After updating dependencies, they introduced conflicting rules. So, ESLint would say that prettier's formatting needs to be corrected and vice versa. To fix this, I updated the prettier script to only format solidity files in the 4337 and added the `examples` directory to `.prettierignore` for the global check (since the only app there uses eslint for formatting as well)

File: modules/4337/test/e2e/run.sh [hunk 2]
```diff
@@ -14,4 +14,4 @@ done
 
 hardhat test --deploy-fixture --network localhost --grep '^E2E - '
 
-docker compose down
+# docker compose down
```
Comment (mmv08) [inline modules/4337/test/e2e/run.sh:17]: Revert bundler changes before merging.

File: modules/4337/docker/bundler/Dockerfile [hunk 3]
```diff
@@ -2,8 +2,10 @@ FROM docker.io/library/node:18
 
 RUN git clone https://github.com/eth-infinitism/bundler /src/bundler
 WORKDIR /src/bundler
-RUN git checkout v0.6.1
+# The latest commit on the v07 branch at the time of writing: https://github.com/eth-infinitism/bundler/commits/v07/
+RUN git checkout 77580829b8bc433d3a818a87a18c3adf3353f675
 
+RUN sed -i 's|0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789|0x506D9aB079BEBB8A25EA07f75e794c0b15f354C2|g' packages/bundler/localconfig/bundler.config.json
```
Comment (mmv08) [inline modules/4337/docker/bundler/Dockerfile:8]: Revert bundler changes before merging.

**Comment by nlordell** [comment]
Closing in favor of #245

**Comment by nlordell** [comment]
Re-opening, as we want to potentially include this to an `entrypoint/v0.7` feature branch.

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
Merging with failing E2E tests (see PR description).
